lib/ingestor-api/runtime/src/collection.py

```diff
@@ -18,12 +19,14 @@ def ingest(collection: StacCollection):
         creds = get_db_credentials(os.environ["DB_SECRET_ARN"])
         with PgstacDB(dsn=creds.dsn_string, debug=True) as db:
             loader = Loader(db=db)
-            collection = [
-                collection.to_dict()
-            ]  # pypgstac wants either a string or an Iterable of dicts.
-            loader.load_collections(file=collection, insert_mode=Methods.upsert)
+            loader.load_collections(
+                file=[collection.model_dump(mode="json")], insert_mode=Methods.upsert
+            )
     except Exception as e:
-        print(f"Encountered failure loading collection into pgSTAC: {e}")
+        raise HTTPException(
```

Comment:
Are there additional exceptions we could raise to help debug?  DB_SECRET_ARN could not be set or get_db_credentials could error out

Reply:
This should catch any error in the whole block and pass a message back to the user. I think it should be a 500 error, though.